ml/rewrite/rewrite.py

```diff
@@ -90,18 +90,27 @@ def decompose_concrete_rewrite_claim(self, provable: ProvableClaim) -> Tuple[kor
     Find and instantiate a rewrite axiom for the given pattern,
     and then resolve all the functions in the RHS
     """
-    def rewrite_from_pattern(self, pattern: kore.Pattern) -> ProvableClaim:
+    def rewrite_from_pattern(self, pattern: kore.Pattern, rewriting_info: List[Tuple[kore.Pattern, kore.Pattern]]) -> ProvableClaim:
         unification_gen = UnificationProofGenerator(self.env)
 
         for _, rewrite_axiom in self.env.rewrite_axioms.items():
             print(f"> trying axiom {KoreTemplates.get_axiom_unique_id(rewrite_axiom.claim)}")
 
             lhs, _, _, _ = self.decompose_rewrite_axiom(rewrite_axiom.claim.pattern)
-            unification_result = unification_gen.unify_patterns(lhs, pattern)
+            print("rewrite_from_pattern", rewriting_info)
```

Comment:
nit: this may be printed multiple times, can we move it outside the loop (or remove it)